crates/runtime/src/accelerated_table/refresh.rs

```diff
@@ -0,0 +1,683 @@
+            check_interval,
+            sql,
+            mode,
+            period,
+        }
+    }
+}
+
+pub(crate) enum AccelerationRefreshMode {
+    Full(Receiver<()>),
+    Append(Option<Receiver<()>>),
+}
+
+pub(crate) struct Refresher {
+    dataset_name: String,
+    federated: Arc<dyn TableProvider>,
+    refresh: Arc<RwLock<Refresh>>,
+    accelerator: Arc<dyn TableProvider>,
+}
+
+impl Refresher {
+    pub(crate) fn new(
+        dataset_name: String,
+        federated: Arc<dyn TableProvider>,
+        refresh: Arc<RwLock<Refresh>>,
+        accelerator: Arc<dyn TableProvider>,
+    ) -> Self {
+        Self {
+            dataset_name,
+            federated,
+            refresh,
+            accelerator,
+        }
+    }
+
+    pub(crate) async fn start(
+        &self,
+        acceleration_refresh_mode: AccelerationRefreshMode,
+        ready_sender: oneshot::Sender<()>,
+    ) {
+        let dataset_name = self.dataset_name.clone();
+        let mut stream = self.stream_updates(acceleration_refresh_mode).await;
+
+        let ctx = SessionContext::new();
+
+        let mut ready_sender = Some(ready_sender);
+
+        loop {
+            let future_result = stream.next().await;
+
+            match future_result {
+                Some(data_update) => {
+                    let Ok(data_update) = data_update else {
+                        continue;
+                    };
+
+                    let state = ctx.state();
+
+                    let overwrite = data_update.update_type == UpdateType::Overwrite;
+                    match self
+                        .accelerator
+                        .insert_into(
+                            &state,
+                            Arc::new(DataUpdateExecutionPlan::new(data_update)),
+                            overwrite,
+                        )
+                        .await
+                    {
+                        Ok(plan) => {
+                            if let Err(e) = collect(plan, ctx.task_ctx()).await {
+                                tracing::error!("Error adding data for {dataset_name}: {e}");
+                            } else if let Some(sender) = ready_sender.take() {
+                                sender.send(()).ok();
+                            };
+                        }
+                        Err(e) => {
+                            tracing::error!("Error adding data for {dataset_name}: {e}");
+                        }
+                    }
+                }
+                None => break,
+            };
+        }
+    }
+
+    async fn stream_updates(
+        &self,
+        acceleration_refresh_mode: AccelerationRefreshMode,
+    ) -> BoxStream<'_, super::Result<DataUpdate>> {
+        let time_column = self.refresh.read().await.time_column.clone();
+
+        match acceleration_refresh_mode {
+            AccelerationRefreshMode::Append(receiver) => {
+                if let (Some(receiver), Some(_)) = (receiver, time_column) {
```

Comment:
This is a bit of a smell - if there isn't a receiver or time column we just ignore it? I think it's fine for now, but I'd want to see this refactored to cover all cases, or make it such that this can't happen

Reply:
Yeah, one reason I was leaving it so is we are going to have more change about OnExchange which is the OG append. So this code may need further changes.